cmd/snap/cmd_known.go

```diff
@@ -110,9 +118,20 @@ func (x *cmdKnown) Execute(args []string) error {
 
 	var assertions []asserts.Assertion
 	var err error
-	if x.Remote {
+	switch {
+	case x.Remote && !x.Direct:
+		// --remote will query snapd
+		assertions, err = x.client.Known(string(x.KnownOptions.AssertTypeName), headers, &client.KnownOptions{Remote: true})
+		// if snapd is unavailable automatically fallback
+		var connErr client.ConnectionError
+		if xerrors.As(err, &connErr) {
+			assertions, err = downloadAssertion(string(x.KnownOptions.AssertTypeName), headers)
+		}
```

Comment:
stylistic question: would inverting the condition and using `fallthrough` be helpful in this case? more DRY but fallthrough is weird so i dunno

Reply:
Not sure TBH